internal/datanode/data_node.go

```diff
@@ -156,6 +169,66 @@ func (node *DataNode) Init() error {
 	return nil
 }
 
+// StartWatchChannels start loop to watch channel allocation status via kv(etcd for now)
+func (node *DataNode) StartWatchChannels(ctx context.Context) {
+	defer logutil.LogPanic()
+	// REF MEP#7 watch path should be [prefix]/channel/{node_id}/{channel_name}
+	watchPrefix := fmt.Sprintf("channel/%d", node.NodeID)
+	evtChan := node.kvClient.WatchWithPrefix(watchPrefix)
+	for {
+		select {
+		case <-ctx.Done():
+			log.Debug("watch etcd loop quit")
+			return
+		case event := <-evtChan:
+			if event.Canceled { // failed to watch
+				log.Warn("Watch channel failed", zap.Error(event.Err()))
+				return
```

Comment:
If returned, DataNode is no longer able to work, Please add a TODO comment here.